### ISSUE TITLE
[spi_host] V1 Signoff 

### ISSUE THREAD

**Issue by msfschaffner** [lowRISC/opentitan#21010]
### Description

Ensure V1 signoff criteria are fulfilled after focus area changes have landed.

> V1 Review Notes [here (OpenTitan ACL)](https://docs.google.com/document/d/1l4dqKUU3TARZLsf1Id1Y5yk1M6SFBvLrGT0Z4mCd8cY)

**Comment by hcallahan-lowrisc** [comment]
Given the small scope of the M2 DD issues, I believe this estimate is excessive. Reducing down to 2. @johngt

**Comment by rswarbrick** [comment]
I've spent some time chatting with @hcallahan-lowrisc and we've had a "mini V1 review" to convince ourselves that the spi_host lifecycle stage hasn't regressed more than V1. Notes [here](https://docs.google.com/document/d/1l4dqKUU3TARZLsf1Id1Y5yk1M6SFBvLrGT0Z4mCd8cY/edit)

We're convinced that we're *almost* at the V1 stage, except that the testplan might need extending to cover recently added features/behaviours. I've opened #22136 to cover that task.

Once that issue is closed, I believe we're good to sign off `spi_host` as still being at V1 level.
